### PR TITLE
WaveShaper Oversampling

### PR DESCRIPTION
**Pull request by dug9**
2x, 4x oversampling implemented from webkit, not tested vs web and doesn't sound much different with / without oversampling

### CODE REVIEW

File: examples/src/Examples.hpp [hunk 1]
```diff
@@ -1645,3 +1645,96 @@ struct ex_split_merge : public labsound_example
     }
     
 };
+// https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/createWaveShaper#example
+
+static void makeDistortionCurveA(std::vector<float> & curve, unsigned n_samples, float amount = .5f)
+{
+    float k = amount * 10.0f;
+    float deg = (float) std::_Pi / 180.0f;
```
Comment (meshula) [inline examples/src/Examples.hpp:1653]: is std::_Pi MSVC specific? Maybe we are stuck with M_PI here for cross platform-ness?

File: src/internal/src/DirectConvolver.cpp [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+//#include "config.h"
+
+//#if ENABLE(WEB_AUDIO)
+
+#include "DirectConvolver.h"
+
+//#if OS(DARWIN)
+//#include <Accelerate/Accelerate.h>
+//#endif
```
Comment (meshula) [inline src/internal/src/DirectConvolver.cpp:37]:
```suggestion
#if OS(DARWIN)
#include <Accelerate/Accelerate.h>
#endif
```
to compile on mac/ios we'll need this

**Review by meshula — CHANGES_REQUESTED** [review]
> doesn't sound much different 

It's good to have an implementation though, so thanks for that. I wonder if there's a use case that demonstrates it in more perceptible way?

There's a couple of small tweaks necessary to get linux/apple compiling (in the notes)

**Comment by dug9** [comment]
Fixed.

File: src/internal/src/DirectConvolver.cpp [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+//#include "config.h"
+
+//#if ENABLE(WEB_AUDIO)
+
+#include "DirectConvolver.h"
+
+#if OS(DARWIN)
```
Comment (meshula) [inline src/internal/src/DirectConvolver.cpp:35]:
```suggestion
#if defined(__APPLE__)
```

File: src/internal/src/DirectConvolver.cpp [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+//#include "config.h"
+
+//#if ENABLE(WEB_AUDIO)
+
+#include "DirectConvolver.h"
+
+#if defined(__APPLE__)
```
Comment (meshula) [inline src/internal/src/DirectConvolver.cpp:35]: #if defined(LABSOUND_PLATFORM_OSX)

**Review by meshula — APPROVED** [review]

**Comment by meshula** [comment]
All platforms are passing! Thanks :)

**Pull request merged**